src/Middleware/RateLimiting/src/RateLimitingExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+/// <summary>
+/// Extension methods for the RateLimiting middleware.
+/// </summary>
+public static class RateLimitingExtensions
+{
+    /// <summary>
+    /// Enables rate limiting for the application.
+    /// </summary>
+    /// <param name="app"></param>
+    /// <returns></returns>
+    public static IApplicationBuilder UseRateLimiting(this IApplicationBuilder app)
+    {
+        if (app == null)
+        {
+            throw new ArgumentNullException(nameof(app));
+        }
+
+        return app.UseMiddleware<RateLimitingMiddleware>();
+    }
+
+    /// <summary>
+    /// Enables rate limiting for the application with the given options.
+    /// </summary>
+    /// <param name="app"></param>
+    /// <param name="options"></param>
+    /// <returns></returns>
+    public static IApplicationBuilder UseRateLimiting(this IApplicationBuilder app, RateLimitingOptions options)
```

Comment:
Should be `Action<RateLimitingOptions>`, follows the pattern we use for other middleware.

Reply:
I got this from StaticFiles based on @halter73's suggestion, but does look like most places use an Action - I'll update

https://github.com/dotnet/aspnetcore/blob/181db595bb11ef99e3a9167e7b940124363c1e36/src/Middleware/StaticFiles/src/StaticFileExtensions.cs#L55

Reply:
Bad static files! We weren't the most consistent on these APIs but are trying to align with the Action<OptionType> and empty overloads.

Reply:
What middleware currently take options the right way? I see UseCors takes a `Action<CorsPolicyBuilder>` which is kinda like `Action<OptionType>`. I vaguely remember trying to switch to callbacks for middleware options but I forget what we've done that for.

Reply:
Discussed this with Brennan, we only really use Actions on `IServiceCollection` extensions.

Reply:
Static files has the app builder options overload because we expect there to be multiple instances. Do we expect there to be multiple instances of this middleware in the pipeline?

Reply:
I expect there easily could be. `MapWhen()` and `UseWhen()` are pretty common. While in most cases you probably should leverage endpoint routing, not everyone does. And there are valid use cases for middleware branching.

Reply:
We stopped supporting this for all new middleware. Let's leave it off until we work out the endpoint routing design. Once the middleware is endpoint routing aware I don't think it will even make sense to have multiple since they'll all be looking at the same endpoint cofing.

Reply:
So the supported way to configure options would be to manually do it on the ServiceCollection? Why not have an extension method for that?

Reply:
Yes, there should be an extension method on IServiceCollection for that.

Reply:
https://github.com/dotnet/aspnetcore/pull/41008/commits/ac0ad26b8006af743d8b12b6760fec18af54ade2